### PR TITLE
 feat(Charts): line and slope charts

### PR DESCRIPTION
**Pull request by tpreusse**
screen shot 2018-02-15 at 18 11 16
screen shot 2018-02-15 at 18 11 28

Also new `showBarValues` feature for bars by @herrstucki

And includes a [React error boundary](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html).
screen shot 2018-02-15 at 19 42 01

### CODE REVIEW

**Comment by tpreusse** [comment]
Merging now for staging, retro review with Dani tomorrow.

**Comment by tpreusse** [comment]
Also review this commit on master: https://github.com/orbiting/styleguide/commit/18f746d6b15a43e2204b3a52f647835f2cf6c119

File: src/components/Chart/Lines.js [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+    .x(d => x(xAccessor(d)))
+    .y0(d => y(d.datum[`confidence${confidence}_lower`]))
+    .y1(d => y(d.datum[`confidence${confidence}_upper`]))
+
+  const linesWithLayout = lines.map(line => {
+    return {
+      ...line,
+      startX: x(xAccessor(line.start)),
+      // we always render at end label outside of the chart area
+      // even if the line end in the middle of the graph
```
Comment (uxengine) [inline src/components/Chart/Lines.js]: Typo: end => ends

File: src/components/Chart/Lines.js [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+    if (domain.length > 5) {
+      let maxIndex = domain.length - 1
+      xTicks = [
+        domain[0],
+        domain[Math.round(maxIndex * 0.33)],
+        domain[Math.round(maxIndex * 0.66)],
+        domain[maxIndex]
+      ].filter(deduplicate)
+    } else {
+      xTicks = domain
```
Comment (uxengine) [inline src/components/Chart/Lines.js]: else-branch is redundant, you already do `xTicks = domain` before the if

File: src/components/Chart/Lines.js [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+
+export const Line = props => <LineChart {...props} />
+
+Line.defaultProps = {
+  x: 'year',
+  xScale: 'time',
+  timeParse: '%Y',
+  timeFormat: '%Y',
+  numberFormat: '.0%',
+  zero: true,
```
Comment (uxengine) [inline src/components/Chart/Lines.js]: Waaaat? ;)
Looks like this triggers a shortened/compact axis. At some point we should find a more meaningful name for this prop. Just saying, OK for now.

File: src/components/Chart/Lines.js [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+      </div>
+    </div>
+  )
+}
+
+LineChart.propTypes = {
+  children: PropTypes.node,
+  values: PropTypes.array.isRequired,
+  width: PropTypes.number.isRequired,
+  mini: PropTypes.bool,
```
Comment (uxengine) [inline src/components/Chart/Lines.js]: Can we have some examples in the .md demonstrating props like this?

**Pull request merged**